tools/shared.py

```diff
@@ -1494,12 +1401,12 @@ def nop(arg):
         if env.get(dangerous) and env.get(dangerous) == non_native.get(dangerous):
           del env[dangerous] # better to delete it than leave it, as the non-native one is definitely wrong
       return env
-    env['CC'] = quote(EMCC) if not WINDOWS else 'python %s' % quote(EMCC)
```

Comment:
why?

Reply:
1. Calling `emcc` internally will cause multiple duplicated Python version warnings
2. I think they are anyway equivalent, is they a specific need to do this conditionally?

Reply:
Yeah, hopefully we don't need the conditional stuff. @juj, was it working around a windows issue?

Reply:
I think it's safe to do this, as we should be calling `[PYTHON, EMCC]` anyhow everywhere, including windows.

Reply:
On Windows it's not possible to invoke `/path/to/emscripten/emcc` when `emcc` is a python script. (there's no concept of reading a shebang from the file on Windows), but one always needs to do `python /path/to/emscripten/emcc` to run the Python script.

That's why we have the `emcc.bat` file on Windows. If one invokes the subprocess with "shell expansion" enabled, then `/path/to/emscripten/emcc` works, and looks up `/path/to/emscripten/emcc.bat`. However shell expansion is sometimes considered poor form in subprocess invocations, which is why we'd prefer to disable shell expansion when invoking, and explicitly call the batch file `/path/to/emscripten/emcc.bat`.

To summarize, on Windows

```py
subprocess.run(['c:/emscripten/emcc`]) # Fails, emcc is a py script, can't execute
subprocess.run(['c:/emscripten/emcc.py`]) # Fails, emcc.py is a py script, can't execute
subprocess.run(['c:/emscripten/emcc.py`], shell=True) # Fails, emcc.py is a py script, can't execute
subprocess.run(['c:/emscripten/emcc`], shell=True) # Works, and invokes emcc.bat
subprocess.run(['c:/emscripten/emcc.bat`], shell=True) # Works, and invokes emcc.bat

subprocess.run(['c:/emscripten/emcc.bat`]) # Works, and invokes emcc.bat, preferred
subprocess.run(['python', 'c:/emscripten/emcc`]) # Works, and invokes emcc python script, slightly more preferred compared to the .bat script to avoid py -> batch -> py roundtripping
subprocess.run(['python', 'c:/emscripten/emcc.py`]) # Works, and invokes emcc.py script, slightly more preferred compared to the .bat script to avoid py -> batch -> py roundtripping
```

We prefer to avoid `shell=True` whenever possible, since it expands PATHs, `%`s and possibly `$`s depending on the calling terminal (PowerShell, MinGW, CygWin, MSYS). Since we're already being explicit with the path where emcc executable resides, it's nicer to explicitly say "emcc.bat is here on this absolute path, call this one", or "run this py script on python".

Reply:
Thanks for great explanation! Yes, we need to explicitly call python on Windows, by *is there a specific need to do this conditionally* I meant that we can also do same thing on non-Windows: `env['CC'] = 'python %s' % quote(EMCC)`

I should have been clearer, sorry.

Reply:
Oh right, I see now I kind of was explaining a separate issue. Yeah, the original adjustment was about https://github.com/kripken/emscripten/commit/bff4d2bc1cc7ef9a44b129978c6920b6b53e6633, and I agree that it should also work on non-Windows, so this change is good.